lib/drive_synchronizer.rb

```diff
@@ -3,22 +3,27 @@ class DriveSynchronizer < Synchronizer
 
     protected
     def perform_sync
-      session = GoogleDrive::Session.from_service_account_key(StringIO.new(@api_key))
+      session = session_meth
+      full_path = backup.path
+      filename = backup.filename
+      file = session.upload_from_file(full_path, filename)
+      add_to_folder(session, file)
+      session.root_collection.remove(file)
+    end
+
+    def session_meth
```

Comment:
You might want to just call this session?

Or you could assign it to an instance variable in the constructor `@session`